src/utils/users.interface.ts

```diff
@@ -0,0 +1,58 @@
+export interface IOrganizationGroup {
+    id?: string;
+    name: string;
+    directManagers?: IUser[] | string[];
+    directMembers?: IUser[] | string[];
+    createdAt: Date;
+    updatedAt?: Date;
+    ancestors?: string[];
+    children?: IOrganizationGroup[] | string[];
+    hierarchy?: string[];
+    isALeaf?: boolean;
+    isAlive?: boolean;
+}
+
+export interface IDomainUser {
+    id?: string;
+    domain?: string;
+    name?: string;
+    uniqueID?: string;
+    adfsUID?: string;
+    personId?: string | IUser;
+}
+
+export interface IUser {
+    // Person's Basic information
+    _id?:string;
+    id: string;
+    identityCard: string;
+    personalNumber?: string;
+    domainUsers: IDomainUser[];
+    entityType: string;
+    serviceType?: string;
+    firstName: string;
+    lastName: string;
+    currentUnit?: string;
+    alive?: boolean;
+    dischargeDay?: Date;
+    hierarchy: string[];
+    hierarchyFlat?: string;
+    directGroup: string | IOrganizationGroup;
+    managedGroup?: string | IOrganizationGroup;
+    rank?: string;
+    updatedAt?: Date;
+    createdAt?: Date;
+    // Editable by the Person
+    job: string;
+    mail?: string;
+    phone?: string[];
+    mobilePhone?: string[];
+    address?: string;
+    // Editable with strong permissions
+    responsibility?: string;
+    responsibilityLocation?: string | IOrganizationGroup;
+    clearance?: string;
+    // Calculated
+    fullName?: string;
+    adfsId: string;
+}
```

Comment:
Why do you need this file if you already have the proto of user-service?

Reply:
@yoavrozner please comment

Reply:
Because I think it is redundant to compile and use compiled typescript files on every proto change instead of changing it in this interface file due to the fact that I use proto-loader package that doesn't require those typescript files. If I go through this long, heavy, time consuming, bug luring, headache giving and unimportant, I will rewrite this service in golang.